Sources/Plasma/FeatureLib/pfPython/pySceneObject.cpp

```diff
@@ -815,9 +816,31 @@ std::vector<PyObject*> pySceneObject::GetPythonMods()
         {   
             for (size_t i = 0; i < obj->GetNumModifiers(); i++)
             {
-                const plPythonFileMod* resp = plPythonFileMod::ConvertNoRef(obj->GetModifier(i));
-                if (resp)
-                    pyPL.push_back(pyKey::New(resp->GetKey()));
+                const plPythonFileMod* pfm = plPythonFileMod::ConvertNoRef(obj->GetModifier(i));
+                if (pfm)
+                    pyPL.push_back(pyKey::New(pfm->GetKey()));
+            }
+        }
+    }
+    return pyPL;
+}
+
+std::vector<PyObject*> pySceneObject::GetImageLibMods()
+{
+    std::vector<PyObject*> pyPL;
+    if (!fSceneObjects.empty())
+    {
+        // get the object pointer of just the first one in the list
+        // (We really can't tell which one the user is thinking of if they are
+        // referring to multiple objects, so the first one in the list will do.)
+        plSceneObject* obj = plSceneObject::ConvertNoRef(fSceneObjects[0]->ObjectIsLoaded());
+        if (obj)
+        {
+            for (size_t i = 0; i < obj->GetNumModifiers(); i++)
+            {
+                const plImageLibMod* ilm = plImageLibMod::ConvertNoRef(obj->GetModifier(i));
+                if (ilm)
+                    pyPL.push_back(pyKey::New(ilm->GetKey()));
```

Comment:
I'm pretty sure that `plSceneObject` has a method for returning modifiers by class index.

Reply:
Darn, I missed that!

Reply:
Indeed.  plSceneObject::GetModifierByType is just what I needed!